openwisp_controller/config/tests/test_vpn.py

```diff
@@ -308,3 +310,35 @@ def test_vpn_and_cert_different_organization(self):
             )
         else:
             self.fail('ValidationError not raised')
+
+    def test_placeholder_dh(self):
+        vpn = self._create_vpn(dh='')
+        self.assertIsNotNone(vpn.dh)
+        self.assertEqual(vpn.dh, Vpn._placeholder_dh)
+
+
+class TestVpnTransaction(
+    TestOrganizationMixin,
+    TestVpnX509Mixin,
+    CreateConfigTemplateMixin,
+    TransactionTestCase,
+):
+    def test_create_vpn_dh_with_vpn_update(self):
+        vpn = self._create_vpn(dh='')
+        with mock.patch.object(create_vpn_dh, 'delay') as mocked_update:
+            vpn.name = 'update_name'
+            vpn.full_clean()
+            vpn.save()
+            mocked_update.assert_not_called()
```

Comment:
this test looks wrong to me. If the dh is empty it will be a falsy value so the `if` block (`if not self.dh:`) in the save method will be triggered and we expect the celery task to be launched.

Reply:
@NoumbissiValere can you please reply on this point?